helix/appendixes/package-principles.rst

```diff
@@ -0,0 +1,131 @@
+Principles of Package and Component Design
+====================================================
+
+Helix is based on Robert Martin's "Principles of Package and Component Design" 
+(Chapter 28,  *Agile Principles, Patterns, and Practices in C#*, Prentice Hall, 2006). 
+These six principles set forth a conceptual framework for managing component boundaries ("cohesion") and interrelationships ("coupling").
+The following discussion presents these principles, provides a brief synopsis of how Martin presents them, and shows how they guide
+specific Helix practices.  Note that the original material uses the term "component," whereas Helix uses the term "module." We will use
+"component" when referring to packages in the original pre-Sitecore context, and we will use "module" when referring to Helix.
+
+The Principles of Component Cohesion
```

Comment:
Isn't the correct name "Principles of Package Cohesion"

Reply:
It's component cohesion in my Kindle copy: 
![image](https://user-images.githubusercontent.com/689532/67247663-10dc6e80-f430-11e9-93d3-7cd959d9685b.png)

Reply:
Martin uses "component" as a generic term, and it is the main term he uses in this chapter. Here he dives in a bit into the distinction.   

![image](https://user-images.githubusercontent.com/689532/67247935-c6a7bd00-f430-11e9-890d-86a38db76bbf.png)

Reply:
Agree the terms are pretty interchangeable, but we should choose one and stick with it.

We're currently linking out to (section 2.1.3) http://butunclebob.com/ArticleS.UncleBob.PrinciplesOfOod which uses the term "package" for the principals.

It's a bit picky I know, just want to be consistent to help people who are new to this? If we do want to start to use the "component" term maybe we need to include the section above to clarify?

Reply:
We also use "package" throughout the Helix course.
I think component could lead people down the wrong path and make them feel that each component could be it's own feature (which I've seen) and therefore becomes too granular.

Reply:
Sounds like there's a consensus here. I will change the language to use "package", and add a sentence or two explaining that both terms are used in Martin's writing, and that the document uses "package" for consistency.

Reply:
Please review changes and note on terminology.